lib/middleware.js

```diff
@@ -1,10 +1,61 @@
 'use strict';
 
+require('dotenv').load();
+const jwt = require('jwt-simple');
+
 module.exports = function(req, res, cb) {
-  /* To be replaced with token validation and/or authentication */
-  const validated = Object.assign(req, {
-    user: 'test-user',
-    isAdmin: false
-  });
-  return cb(null, validated);
+  const token = getToken(req);
+  let payload;
+  if (token) {
+    payload = getPayload(token);
+  } else {
+    payload = null;
+  }
+
+  // QUESTION: should we check if the user if belongs to the list
+  // of authorized users here and add like an `.isTrustedUser` property or
+  // so to the request?
+  if (payload) {
+    req.user = payload;
+    if (process.env.trustedUsers) {
+      const trustedUsers = JSON.parse(process.env.trustedUsers);
+      if (trustedUsers.indexOf(payload.id) !== -1) {
+        req.isTrustedUser = true;
+      } else {
+        req.isTrustedUser = false;
+      }
+    } else {
+      req.isTrustedUser = true; // if no env var, all OSM users are trusted
+    }
+  } else {
+    req.user = null;
+    req.isTrustedUser = false;
+  }
+
+  return cb();
 };
+
+/**
+ * Gets the token from the Authorization header if it exists
+ *
+ * @param request {RequestObject} - Express request object
+ * @returns {String} token extracted from Authorization header 
+ */
+function getToken(request) {
+  if (!request.headers.Authorization) return null;
+  const authHeader = request.headers.Authorization;
+  const token = authHeader.replace('Token ', '');
+  //TODO: validate that the token is a valid JWT token
```

Comment:
I think we can either check that it's a valid JWT token here, or in the `getPayload` function. To me, it makes sense for this function's objective to be to extract the token, if it exists, from the request headers, and the `getPayload` function's objective to handle the decoding and any errors associated with that.